src/gui/sdlmain.cpp

```diff
@@ -956,36 +1011,38 @@ Bitu GFX_SetSize(Bitu width, Bitu height, Bitu flags,
 		changing between modes with different dimensions */
 		SDL_FillRect(sdl.surface, NULL, SDL_MapRGB(sdl.surface->format, 0, 0, 0));
 		SDL_UpdateWindowSurface(sdl.window);
-		break;
+		sdl.desktop.type = SCREEN_SURFACE;
+		break; // SCREEN_SURFACE
+
 	case SCREEN_TEXTURE: {
-		int imgw, imgh, wndw, wndh; /* image and window width and height  */
 		/* TODO: set up all ScalingMode-related settings here. Currently, the */
 		/*       interpolation hint is set at the reading of settings.    */
 		if (sdl.scaling_mode != SmPerfect) {
-			if (!GFX_SetupWindowScaled(sdl.desktop.want_type)) {
-				LOG_MSG("SDL:Can't set video mode, falling back to surface");
+			if (!SetupWindowScaled(SCREEN_TEXTURE, false)) {
+				LOG_MSG("SDL: Can't set video mode, falling "
+				        "back to surface");
 				goto dosurface;
 			}
 		} else {
-			imgw = sdl.ppscale_x * sdl.draw.width;
-			imgh = sdl.ppscale_y * sdl.draw.height;
+			const int imgw = sdl.ppscale_x * sdl.draw.width;
+			const int imgh = sdl.ppscale_y * sdl.draw.height;
 
+			int wndw, wndh;
 			if (sdl.desktop.fullscreen) {
-				wndh = avh;  wndw = avw;
+				wndh = avh;
+				wndw = avw;
 			} else {
-				wndh = imgh; wndw = imgw; 
+				wndh = imgh;
+				wndw = imgw;
 			}
 
 			sdl.clip.w = imgw;
 			sdl.clip.h = imgh;
 			sdl.clip.x = (wndw - imgw) / 2;
 			sdl.clip.y = (wndh - imgh) / 2;
 
-
-			sdl.window = GFX_SetSDLWindowMode(
-				wndw, wndh, sdl.desktop.fullscreen, 
-				SCREEN_TEXTURE);
-
+			sdl.window = SetWindowMode(SCREEN_TEXTURE, wndw, wndh,
+			                           sdl.desktop.fullscreen, false);
```

Comment:
Suggest using a placeholder bool just to make calls to `SetWindowMode` a bit more readable and explicit:

```
const bool is_resizeable = false;
sdl.window = SetWindowMode(SCREEN_TEXTURE, wndw, wndh,
			                           sdl.desktop.fullscreen, is_resizeable);
```
(and other calls below)

Reply:
Maybe an enum would be even more readable, especially if used in a handful of places?

Reply:
I went with `constexpr bool FIXED_SIZE = false`: 5927f4cf7ea0e8478e1958bcea6226c0619f353a - it communicates the intention and compiler will remind us when it'll be time to remove it :)